x-pack/plugins/ml/public/application/ml_nodes_check/check_ml_nodes.ts

```diff
@@ -6,15 +6,21 @@
  */
 
 import { ml } from '../services/ml_api_service';
+import type { MlApiServices } from '../services/ml_api_service';
 import type { MlNodeCount } from '../../../common/types/ml_server_info';
 
 let mlNodeCount: number = 0;
 let lazyMlNodeCount: number = 0;
 let userHasPermissionToViewMlNodeCount: boolean = false;
 
-export async function getMlNodeCount(): Promise<MlNodeCount> {
+export async function getMlNodeCount(mlApiServices?: MlApiServices): Promise<MlNodeCount> {
+  let nodes;
   try {
-    const nodes = await ml.mlNodeCount();
+    if (mlApiServices) {
+      nodes = await mlApiServices.mlNodeCount();
+    } else {
+      nodes = await ml.mlNodeCount();
+    }
```

Comment:
```suggestion
const nodes = await (mlApiServices ?? ml).mlNodeCount();
```
To remove the `let`

Reply:
We should make a hook version of this so we can get the service from the context

Reply:
Good idea, I'll add it as part of the follow up refactor work 👍